gamemodes/terrortown/gamemode/server/sv_main.lua

```diff
@@ -841,37 +841,41 @@ function TellTraitorsAboutTraitors()
 		end
 	end
 
-	hook.Run("TTT2TellTraitors", traitornicks)
+	for _, v in ipairs(player.GetAll()) do
+		if v:HasTeam(TEAM_TRAITOR) then
 
-	-- This is ugly as hell, but it's kinda nice to filter out the names of the
-	-- traitors themselves in the messages to them
-	local traitornicks_min = #traitornicks < 2
+			local tmp, shouldShow = table.Copy(traitornicks), true
 
-	if not traitornicks_min then
-		table.sort(traitornicks, function(a, b)
-			return a and b and a:upper() < b:upper()
-		end)
-	end
+			tmp, shouldShow = hook.Run("TTT2TellTraitors", tmp, v)
 
-	for _, v in ipairs(player.GetAll()) do
-		if v:HasTeam(TEAM_TRAITOR) then
-			if traitornicks_min then
-				LANG.Msg(v, "round_traitors_one")
+			if shouldShow == nil then shouldShow = true end
+			if shouldShow == false then continue end
+
+			if tmp == nil then tmp = table.Copy(traitornicks) end
```

Comment:
we don't need a return

Reply:
tmp is modified if the hook is called

Reply:
https://github.com/TTT-2/TTT2/pull/361/commits/ad2d57d01c1b959d461464993c872474c46828fd
I mean, I will get there eventually

Reply:
if the table is nil, this table was set to nil manually. This shouldn't be reverted, there should be a continue if you wanna check this :)